production/helm/loki/templates/gateway/deployment-gateway.yaml

```diff
@@ -32,6 +32,13 @@ spec:
         {{- include "loki.gatewaySelectorLabels" . | nindent 8 }}
     spec:
       serviceAccountName: {{ include "loki.serviceAccountName" . -}}
+      {{- if semverCompare ">=1.13-0" .Capabilities.KubeVersion.GitVersion }}
+      {{- if or (.Values.loki.enableServiceLinks) (eq (.Values.loki.enableServiceLinks | toString) "<nil>") }}
```

Comment:
why doe we need the `toString` compared to `"<nil>"`? shouldn't helm treat a nil value as falsey? Also, would you consider moving this to a single `if` statement, potentially even extracting a template function if it's messy?

Reply:
The `toString` is to set the `enableServiceLinks` to true (default value) when `enableServiceLinks` is not provided in the values.yaml.

We will need  to have 2 if because in k8s version < v1.13.0, there is no `enableServiceLinks` and I have moved this chunk into template to make it dry.

Reply:
We were deploying the loki-stack in the same namespace as knative and as the service grow, we were not able to start the pod due to too many services environment variables being injected. We are moving loki-stack to different namespace at the moment to fix the issue, but thought this feature could help those that has constraint with the namespace(ie. only allow to deploy to certain namespace).

You can refer to the link I have attached about the pod unable to startup due to too many service environments injected into the pods.
Here is the link for an interesting experiment did for the knative: https://github.com/knative/serving/issues/8498